src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -5257,6 +5267,274 @@ def remove_github_actions(cmd, resource_group, name, repo, token=None, slot=None
+                                    "Please check that the app is a part of the current subscription" % name)
+    current_rg = app_details.resource_group
+    if resource_group is not None and (resource_group.lower() != current_rg.lower()):
+        raise ValidationError("The webapp %s exists in ResourceGroup %s and does not match "
+                              "the value entered %s. Please re-run command with the correct "
+                              "parameters." % (name, current_rg, resource_group))
+
+    # Verify github repo
+    from github import Github, GithubException
+    from github.GithubException import BadCredentialsException, UnknownObjectException
+
+    if repo.strip()[-1] == '/':
+        repo = repo.strip()[:-1]
+
+    g = Github(token)
+    github_repo = None
+    try:
+        github_repo = g.get_repo(repo)
+        try:
+            github_repo.get_branch(branch=branch)
+        except GithubException as e:
+            error_msg = "Encountered GitHub error when accessing {} branch in {} repo.".format(branch, repo)
+            if e.data and e.data['message']:
+                error_msg += " Error: {}".format(e.data['message'])
+            raise CLIError(error_msg)
+        logger.warning('Verified GitHub repo and branch')
+    except BadCredentialsException:
+        raise ValidationError("Could not authenticate to the repository. Please create a Personal Access Token and use "
+                              "the --token argument. Run 'az webapp deployment github-actions add --help' "
+                              "for more information.")
+    except GithubException as e:
+        error_msg = "Encountered GitHub error when accessing {} repo".format(repo)
+        if e.data and e.data['message']:
+            error_msg += " Error: {}".format(e.data['message'])
+        raise CLIError(error_msg)
+
+    # Check if workflow exists in repo and remove
+    file_name = "{}_{}({}).yml".format(
+        branch.replace('/', '-'), name.lower(), slot) if slot else "{}_{}.yml".format(
+            branch.replace('/', '-'), name.lower())
+    dir_path = "{}/{}".format('.github', 'workflows')
+    file_path = "{}/{}".format(dir_path, file_name)
+    existing_publish_profile_name = None
+    try:
+        existing_workflow_file = github_repo.get_contents(path=file_path, ref=branch)
+        existing_publish_profile_name = _get_publish_profile_from_workflow_file(
+            workflow_file=str(existing_workflow_file.decoded_content))
+        logger.warning("Removing the existing workflow file")
+        github_repo.delete_file(path=file_path, message="Removing workflow file, disconnecting github actions",
+                                sha=existing_workflow_file.sha, branch=branch)
+    except UnknownObjectException as e:
+        error_msg = "Error when removing workflow file."
+        if e.data and e.data['message']:
+            error_msg += " Error: {}".format(e.data['message'])
+        raise CLIError(error_msg)
```

Comment:
Could we use the specific error type `FileOperationError` instead of `CLIError`?